angular/src/pipes/color-password-count.pipe.ts

```diff
@@ -0,0 +1,24 @@
+import { Pipe } from '@angular/core';
+import { ColorPasswordPipe } from './color-password.pipe';
+
+/*
+ An updated pipe that extends ColourPasswordPipe to include a character count
+*/
+@Pipe({ name: 'colorPasswordCount' })
+export class ColorPasswordCountPipe extends ColorPasswordPipe {
+    transform(password: string) {
+        const passwordArray = Array.from(password);
```

Comment:
It looks like this logic is the same as the base class. If the only difference is the template literal, could you just make the template literal a property that you override in the subclass, and leave the logic unchanged?

(That said - I like the `getCharacterAndType` refactor either way, it makes everything more readable)